packages/starlight/schemas/i18n.ts

```diff
@@ -162,3 +166,21 @@ function pagefindI18nSchema() {
 		})
 		.partial();
 }
+
+function expressiveCodeI18nSchema() {
+	return z
+		.object({
+			'expressiveCode.copyButtonCopied': z
+				.string()
+				.describe('Expressive Code UI translation. English default value: `"Copied!"`'),
+
+			'expressiveCode.copyButtonTooltip': z
+				.string()
+				.describe('Expressive Code UI translation. English default value: `"Copy to clipboard"`'),
+
+			'expressiveCode.terminalWindowFallbackTitle': z
+				.string()
+				.describe('Expressive Code UI translation. English default value: `"Terminal window"`'),
```

Comment:
Unless I’m mistaken Expressive Code only provides these in English and German currently. Should Starlight look to add these in their built-in translation `.json` files given we’re likely supporting more languages?

Reply:
I was just following the example set by the pagefind schema here. The idea was to have these strings translatable through the Starlight `.json` translation files, and my understanding was that this is achieved by the code in this PR. I also tested this locally, however this was before your changes to the i18n logic. :)

Reply:
I just tested it again and it still works. I added an entry `"expressiveCode.copyButtonCopied": "That worked!"` to `packages\starlight\translations\en.json` and it gets displayed. I also tested it with the Spanish version and it's also possible to customize the string there. Seems to work as intended!

Reply:
Ah sorry, I think my comment wasn’t clear — I also tested this and it was working great!

My question was more about built-in support for more languages. From what I could see in the EC repo, only German and English strings were provided there, so I was asking whether we should be asking translators to add translations to Starlight’s built-in JSON files or upstream to EC — all other parts of the UI are available in 20+ languages, so would be good to work towards (once this PR is merged) also having EC strings covered in those languages.

Reply:
Ahh, thanks a lot for clarifying! I would love to have translations for the EC strings in all these beautiful languages, but I didn't want to add the burden on Starlight translators to also submit PRs to my upstream EC repository. :)

Therefore, my current plan was to let translators keep following the Starlight-focused workflow they are used to, and perform the work of copying the translations to EC at some point myself. I was hoping that this is the easiest process for the translators.

Reply:
Sounds good. We can also encourage our translators to contribute to EC. I might recommend moving your strings out of the code like they are now into JSON or similar files that are easy to find and edit to help people contribute there.

Reply:
I agree! After seeing how i18n is done in Starlight, I was also thinking about moving the translations into separate files in EC as well. That's maybe another reason why it's better to have translators focus on the Starlight JSON files for now until I've made the translation process in my upstream repo easier.